catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/sharing/ShareableMetacardAttributes.java

```diff
@@ -11,51 +11,55 @@
  * License is distributed along with this program and can be found at
  * <http://www.gnu.org/licenses/lgpl.html>.
  */
-package org.codice.ddf.catalog.ui.forms.data;
+package org.codice.ddf.catalog.ui.metacard.sharing;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.MetacardType;
 import ddf.catalog.data.impl.AttributeDescriptorImpl;
 import ddf.catalog.data.impl.BasicTypes;
-import ddf.catalog.data.impl.MetacardTypeImpl;
 import ddf.catalog.data.types.Core;
+import java.util.Map;
 import java.util.Set;
 
-/**
- * Set of attributes needed to enable sharing on Metacards. Should be refactored to live in a more
- * central location at a later time.
- *
- * <p><i>This code is experimental. While it is functional and tested, it may change or be removed
- * in a future version of the library.</i>
- *
- * <p>TODO DDF-3671 Revisit sharing functionality for metacards
- */
-@SuppressWarnings("squid:S1135" /* Action to-do has a ticket number and will be addressed later */)
-public class ShareableAttributes extends MetacardTypeImpl {
-  public static final String SHAREABLE_TAG = "shareable";
+public class ShareableMetacardAttributes implements MetacardType {
```

Comment:
❗️ To be truly generic, these attributes need to be incorporated into the types in catalog-core-api. That includes following all of our other taxonomy conventions and working through ramifications for other potential sharing types. See overall review comment for summary.

Reply:
Eventually I could see these moving there but I think it is ok to continue to incubate them in the Intrigue project.